crates/rnote-engine/src/fileformats/rnoteformat/mod.rs

```diff
@@ -23,18 +23,31 @@ use std::io::{Read, Write};
 
 /// Compress bytes with gzip.
 fn compress_to_gzip(to_compress: &[u8]) -> Result<Vec<u8>, anyhow::Error> {
-    let mut encoder =
-        flate2::write::GzEncoder::new(Vec::<u8>::new(), flate2::Compression::default());
+    let mut encoder = flate2::write::GzEncoder::new(Vec::<u8>::new(), flate2::Compression::new(5));
     encoder.write_all(to_compress)?;
     Ok(encoder.finish()?)
 }
 
 /// Decompress from gzip.
 fn decompress_from_gzip(compressed: &[u8]) -> Result<Vec<u8>, anyhow::Error> {
+    // Optimization for the gzip format, defined by RFC 1952
+    // capacity of the vector defined by the size of the uncompressed data
+    // given in little endian format, by the last 4 bytes of "compressed"
+    //
+    //   ISIZE (Input SIZE)
+    //     This contains the size of the original (uncompressed) input data modulo 2^32.
+    let mut bytes: Vec<u8> = {
+        let mut decompressed_size: [u8; 8] = [0; 8];
```

Comment:
Instead of hardcoding this as 8 bytes and using `usize::from_le_bytes` I think [u32::from_le_bytes](https://doc.rust-lang.org/stable/core/primitive.u32.html#method.from_le_bytes) should be used and then be cast to usize. Otherwise this would fail to compile on 32 bit architectures because then `usize::from_le_bytes` expects an array with size 4 (see [docs](https://doc.rust-lang.org/stable/core/primitive.usize.html#method.from_le_bytes))

Reply:
Good catch, on it